vars/infra.groovy

```diff
@@ -116,7 +116,7 @@ Object withArtifactCachingProxy(Closure body) {
   }
 
   // If the build concerns a pull request, check if there is "skip-artifact-caching-proxy" label applied in case the user doesn't want ACP
-  if ((useArtifactCachingProxy) && (!env.BRANCH_IS_PRIMARY)) {
+  if ((useArtifactCachingProxy) && (!env.BRANCH_IS_PRIMARY) && (env.CHANGE_URL)) {
```

Comment:
```suggestion
  if ((useArtifactCachingProxy) && (env.CHANGE_URL)) {
```

Reply:
Addressed in https://github.com/jenkins-infra/pipeline-library/pull/595